resolwe/test/testcases/__init__.py

```diff
@@ -99,66 +55,78 @@ def _pre_setup(self, *args, **kwargs):
-            flow_executor_settings.get('CONTAINER_NAME_PREFIX', 'resolwe'),
-            get_random_string(length=6),
-            os.path.basename(flow_executor_settings['DATA_DIR'])
-        )
-
-        self.settings = override_settings(FLOW_EXECUTOR=flow_executor_settings,
-                                          FLOW_DOCKER_MAPPINGS=flow_docker_mappings)
-
     def _clean_up(self):
         """Clean up after test."""
         if not self._keep_data:
+            # Do delete this here. See comment below near the makedirs
+            # in setUp.
             shutil.rmtree(settings.FLOW_EXECUTOR['DATA_DIR'], ignore_errors=True)
             shutil.rmtree(settings.FLOW_EXECUTOR['UPLOAD_DIR'], ignore_errors=True)
+            shutil.rmtree(settings.FLOW_EXECUTOR['RUNTIME_DIR'], ignore_errors=True)
 
         self.settings.disable()
 
     def setUp(self):
         """Prepare environment for test."""
         super(TestCaseHelpers, self).setUp()
 
-        self._override_executor_settings()
+        # Override flow executor settings
+        flow_executor_settings = copy.copy(getattr(settings, 'FLOW_EXECUTOR', {}))
+
+        # Get data directory settings
+        data_dir, upload_dir, runtime_dir = (
+            FLOW_EXECUTOR_SETTINGS['DATA_DIR'],
+            FLOW_EXECUTOR_SETTINGS['UPLOAD_DIR'],
+            FLOW_EXECUTOR_SETTINGS['RUNTIME_DIR'],
+        )
+        # Directories need to be recreated here in case a previous
+        # TestCase deleted them. Moving this logic into the test runner
+        # and manager infrastructure would not work, because the manager
+        # and listener can't know where the testcase boundaries are,
+        # they just see a series of data objects; deleting too soon
+        # might cause problems for some tests. The runner does not have
+        # any code between tests, so could only remove data at the very
+        # end, by which time it's already too late, since some tests may
+        # deal specifically with the purging functionality and should
+        # start in a clean environment, without the sediment
+        # (e.g. jsonout.txt, stdout.txt) from previous tests.
+        os.makedirs(data_dir, exist_ok=True)
+        os.makedirs(upload_dir, exist_ok=True)
+        os.makedirs(runtime_dir, exist_ok=True)
+
+        # Override container name prefix setting
+        flow_executor_settings['CONTAINER_NAME_PREFIX'] = '{}_{}_{}'.format(
+            getattr(settings, 'FLOW_EXECUTOR', {}).get('CONTAINER_NAME_PREFIX', 'resolwe'),
+            # NOTE: This is necessary to avoid container name clashes when tests are run from
+            # different Resolwe code bases on the same system (e.g. on a CI server).
+            get_random_string(length=6),
+            os.path.basename(data_dir)
+        )
+
+        # Override Docker data directory mappings
+        flow_docker_mappings = copy.copy(getattr(settings, 'FLOW_DOCKER_MAPPINGS', []))
```

Comment:
Why would you need to update `FLOW_DOCKER_MAPPINGS` here if `FLOW_EXECUTOR['DATA_DIR']`, ... were not updated in `setUp()`?